src/words.ts

```diff
@@ -0,0 +1,161 @@
+type StepVector = Record<string, number>;
+
+// Helper function for the mathematical mod `m` operator to make it return nonnegative integers less than `m`.
+function mod(a: number, m: number): number {
+  return ((a % m) + m) % m;
+}
+
+function sum(array: number[]) {
+  let sum = 0;
+  array.forEach(x => {
+    sum += x;
+  });
+  return sum;
+}
+
+// Helper function that sorts a `Record` in place by key.
+function sortRecordInPlace(record: Record<string, number>) {
+  record = Object.keys(record)
```

Comment:
There's no such thing as an ordered record/object in JavaScript. Also this assignment statement has no effect.

Reply:
> There's no such thing as an ordered record/object in JavaScript. Also this assignment statement has no effect.

When I call this:
```
// Return the set of letters.
export function letters(sv: StepVector): string[] {
  return Object.keys(sv);
}
```

will the array of letters come out sorted?

Reply:
They will not. A oneliner would be `Array.from(Object.keys(sv)).sort()` because JS doesn't care about voiding mutating calls.